topology/probes/libvirt/libvirtgo.go

```diff
@@ -0,0 +1,141 @@
+// +build libvirt,linux
+
+/*
+ * Copyright (C) 2018 Orange.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package libvirt
+
+import (
+	"context"
+	"fmt"
+
+	libvirtgo "github.com/libvirt/libvirt-go"
+	"github.com/skydive-project/skydive/logging"
+)
+
+type libvirtgoDomain struct {
+	libvirtgo.Domain
+}
+
+func (d libvirtgoDomain) GetName() (string, error) {
+	return d.Domain.GetName()
+}
+func (d libvirtgoDomain) GetXML() ([]byte, error) {
+	xml, err := d.Domain.GetXMLDesc(0)
+	return []byte(xml), err
+}
+
+func (d libvirtgoDomain) GetState() (DomainState, int, error) {
+	state, i, err := d.Domain.GetState()
+	return DomainState(state), i, err
+}
+
+type LibvirtgoMonitor struct {
+	*libvirtgo.Connect
+	cidLifecycle int // libvirt callback id of monitor to unregister
+	cidDevAdded  int // second monitor on devices added to domains
+}
+
+func (m *LibvirtgoMonitor) AllDomains() ([]domain, error) {
+	domains, err := m.ListAllDomains(0)
+	if err != nil {
+		return nil, err
+	}
+	allDomains := make([]domain, len(domains))
+	for i, domain := range domains {
+		allDomains[i] = &libvirtgoDomain{domain}
+	}
+	return allDomains, nil
+}
+
+func (m *LibvirtgoMonitor) Stop() {
+	if m.cidLifecycle != -1 {
+		if err := m.DomainEventDeregister(m.cidLifecycle); err != nil {
+			logging.GetLogger().Errorf("Problem during deregistration: %s", err)
+		}
+		if err := m.DomainEventDeregister(m.cidDevAdded); err != nil {
+			logging.GetLogger().Errorf("Problem during deregistration: %s", err)
+		}
+	}
+
+	if _, err := m.Close(); err != nil {
+		logging.GetLogger().Errorf("Problem during close: %s", err)
+	}
+}
+
+func newMonitor(ctx context.Context, probe *Probe) (*LibvirtgoMonitor, error) {
+	if probe.uri == "" {
+		probe.uri = "qemu:///system"
```

Comment:
You may consider making this configurable: https://libvirt.org/uri.html#URI_qemu

Reply:
This is configurable as `uri` comes from the configuration file. If it's empty - and for the libvirt-go based implementation - we use fallback to `qemu:///system`